homeassistant/components/flume/config_flow.py

```diff
@@ -33,45 +33,57 @@
 )
 
 
-async def validate_input(hass: core.HomeAssistant, data):
+def _validate_input(hass: core.HomeAssistant, data: dict, clear_token_file: bool):
+    """Validate in the executor."""
+    flume_token_full_path = hass.config.path(
+        f"{BASE_TOKEN_FILENAME}-{data[CONF_USERNAME]}"
+    )
```

Comment:
A general question: Is it a good pattern to store the pattern in the filesystem, or are there better ways?

Reply:
Ideally we would store the token in the config entry by modelling the auth library as mentioned in our docs https://developers.home-assistant.io/docs/api_lib_auth

But if we are going to use a file, I suggest we store it in `.storage`

Reply:
However, I also think that it's out of scope for this PR, as this is an existing path.

Reply:
It would definitely better to not hit the filesystem here as we want to abstract away the fact that it exists... but that is how the upstream library works in this case.